Makefile

```diff
@@ -0,0 +1,290 @@
+.PHONY: help
+# target: help - Print this help
+help:
+	@egrep "^# target: " Makefile \
+		| sed -e 's/^# target: //g' \
+		| sort \
+		| awk '{printf("    %-20s", $$1); $$1=$$2=""; print "-" $$0}'
+
+.PHONY: build
+# target: build - Build package, run tests and create distribution
+build: .asdf check-epmd
+	@mvn
+
+.PHONY: clouseau1
+# target: clouseau1 - Start local inistance of clouseau1 node
+clouseau1: .asdf check-epmd
+	@mvn scala:run -Dname=$@
+
+.PHONY: clouseau2
+# target: clouseau2 - Start local inistance of clouseau2 node
+clouseau2: .asdf check-epmd
+	@mvn scala:run -Dname=$@
+
+.PHONY: clouseau3
+# target: clouseau3 - Start local inistance of clouseau3 node
+clouseau3: .asdf check-epmd
+	@mvn scala:run -Dname=$@
+
+.PHONY: clean
+# target: clean - Remove build artifacts
+clean:
+	@mvn clean
+
+check-epmd:
+	@if ! lsof -i TCP:4369 | grep -q epmd ; then echo "Error: Please start 'epmd' first"; exit 1; fi
+
+# ==========================
+# Setting up the environment
+# --------------------------
+
+define CURRENT_SHELL
+"$(shell basename $${SHELL})"
+endef
+
+define CURRENT_OS
+"$(shell uname -s | tr '[:upper:]' '[:lower:]')"
+endef
+
+# Testing from fish
+# set SHELL bash && make .asdf
+# set SHELL fish && make .asdf
+# set SHELL zsh && make .asdf
+# set SHELL unsupported && make .asdf
+# Testing from other shells
+# SHELL=bash && make .asdf
+# SHELL=fish && make .asdf
+# SHELL=zsh && make .asdf
+# SHELL=unsupported && make .asdf
+
+ifndef ASDF_DIRENV_BIN
+.asdf:
+	@case $(CURRENT_SHELL)-$(CURRENT_OS) in \
+       fish-darwin) echo "$$FISH_DARWIN_HELP" ;; \
+       bash-darwin) echo "$$BASH_DARWIN_HELP" ;; \
+       zsh-darwin) echo "$$ZSH_DARWIN_HELP" ;; \
+       *) echo "$$CONTRIBUTE_SHELL_SETUP" ;; \
+    esac
+else
+.asdf:
+	@touch $@
+	@touch .envrc
+endif
+
+define FISH_DARWIN_HELP
+There are two steps which need to be done to set up environment.
+
+  -  The asdf tool manager would need to be installed
+  -  The direnv plugin for asdf manager would need to be installed
+
+
+= Setting up asdf tool manager
+
+  You can install asdf from brew by following command.
+
+  ```
+  brew install asdf
```

Comment:
Mind that it is specific to Homebrew (technically macOS), `asdf` could be installed in other ways for different environments.